crates/indexer/src/models/token_models/token_activities.rs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use super::token_utils::{TokenDataIdType, TokenEvent};
+use crate::schema::token_activities;
+use aptos_api_types::{Event as APIEvent, Transaction as APITransaction};
+use bigdecimal::{BigDecimal, Zero};
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Queryable, Serialize)]
+#[diesel(primary_key(
+    transaction_version,
+    event_account_address,
+    event_creation_number,
+    event_sequence_number
+))]
+#[diesel(table_name = token_activities)]
+pub struct TokenActivity {
+    pub transaction_version: i64,
+    pub event_account_address: String,
+    pub event_creation_number: i64,
+    pub event_sequence_number: i64,
+    pub token_data_id_hash: String,
+    pub property_version: BigDecimal,
+    pub creator_address: String,
+    pub collection_name: String,
+    pub name: String,
+    pub transfer_type: String,
```

Comment:
why do we have transfer_type in general token activity table?

Reply:
should it be more like` activity name`, ` activity_parameters`?downstream apps can parse it based on the activity name.

Reply:
what's activity_parameters? For ease of use we don't want to have json blobs so structured data is the way to go even though we lose some flexibility. In that case activity_name or transfer_type is just semantics. I find activity name to be a bit too literal. Transfer type is not exactly accurate but is more in line with something I'd see in an external API.